### ISSUE TITLE
Alternative installation directories

### ISSUE THREAD

**Issue by bitterbutt** [brianmiller/phvalheim-client#1]
If Valheim is installed on another drive but Steam was initially installed on a different drive, the PhValheim client fails to locate the installation directory.

For example:
Steam is installed on the C: drive
Valheim is installed on the D: drive
SteamGetter in steam.cs only checks the registry for where Steam is installed (ex. c:/program files (x86)/steam/steam.exe)
The client then copies the mods and such to the C: drive, but launches Valheim by appId from the actual install location

To fix:
Search the following registry key: HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Steam App 892970
Take the InstallLocation entry (ex. D:\Program Files (x86)\Steam\steamapps\common\Valheim)

Sorry, I would submit a fix for this, but I am not proficient in C# outside of reading it. I'll try and make something anyhow, but wanted to make this none.

**Comment by bitterbutt** [comment]
Here is what ChatGPT spit out for an example.

```
using Microsoft.Win32;

namespace ConsoleApp
{
    class Program
    {
        static void Main(string[] args)
        {
            const string keyName = @"HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Steam App 892970\InstallLocation";

            // Open the specified registry key
            var key = Registry.LocalMachine.OpenSubKey(keyName);

            // Get the value of the "InstallLocation" registry key
            var installLocation = key.GetValue("InstallLocation");

            // Print the value of the "InstallLocation" registry key to the console
            Console.WriteLine(installLocation);
        }
    }
}
```

**Comment by Swesen** [comment]
I've been testing moving the installation to another drive it still works for me, the problem might stem from the assumption that C drive has more than 4GB of space to download the full map and mods since it uses appdata as storage.

I've had two people who had problems joining my server, never had the issue myself so these are just assumptions.

**Comment by brianmiller** [comment]
Thanks guys.  I'll see what I can do to add more intelligence to the client.

**Comment by brianmiller** [comment]
Another report of this being an issue in our Discord. 

This will be the next item I work on after the holidays.

-Brian

**Comment by brianmiller** [comment]
This took more work than I expected.  The registry doesn't hold any keys indicating the location of a game. I.e., Steam library the game was installed into.

`HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall` captures the location the game was installed to but is only updated once the game is uninstalled.

Commit https://github.com/brianmiller/phvalheim-client/commit/c0608824afd340b5b9518a70b8d9930b7b1c87b8 solves the issue with custom logic by using the .vdf and .acf files that the Steam client creates and updates during library population.

I've tested this logic with local disks, local USB drives and network mounted SMB shares. All work as a Steam Library and all now work with PhValheim.  I've also tested migration scenarios. e.g., "Move install folder..."

All download links "Download PhValheim Client" in everyone's public PhValheim UI will now download this new version.

-Brian
